test/performance/performance.go

```diff
@@ -0,0 +1,333 @@
+func printTable(data []metricData) {
+	table := tabwriter.NewWriter(os.Stdout, 1, 4, 1, ' ', tabwriter.Debug|tabwriter.AlignRight)
+	fmt.Fprintln(table, "========\t==========\t====================\t====================\t===================\t===================\t")
+	fmt.Fprintln(table, "time\t# policies\tavg cpu (controller)\tmax cpu (controller)\tavg cpu (apiserver)\tmax cpu (apiserver)\t")
+	fmt.Fprintln(table, "========\t==========\t====================\t====================\t===================\t===================\t")
+
+	for i := 0; i < len(data); i++ {
+		fmt.Fprintf(table, "%s\t%d\t%s\t%s\t%s\t%s\t\n",
+			data[i].timestamp,
+			data[i].numPolicies,
+			fmt.Sprintf("%.5f", data[i].controllerAvg),
+			fmt.Sprintf("%.5f", data[i].controllerMax),
+			fmt.Sprintf("%.5f", data[i].apiserverAvg),
+			fmt.Sprintf("%.5f", data[i].apiserverMax),
+		)
+	}
+
+	table.Flush()
+}
+
+// export table of results to a csv file
+func exportTable(data []metricData, filename string) {
+	f, err := os.Create(filename)
+	if err != nil {
+		klog.Exitf("Error: failed to create %s; %s", filename, err)
+	}
+
+	defer func() {
+		if cerr := f.Close(); cerr != nil && err == nil {
+			err = cerr
+		}
+	}()
+
+	w := csv.NewWriter(f)
+	defer w.Flush()
+
+	line := []string{"time", "num_policies", "avg_cpu_controller", "max_cpu_controller", "avg_cpu_apiserver", "max_cpu_apiserver"}
+	if err := w.Write(line); err != nil {
+		klog.Exitf("Error writing headers to file; %s", err)
+	}
+
+	for _, entry := range data {
+		line = []string{
+			entry.timestamp,
+			fmt.Sprintf("%d", entry.numPolicies),
+			fmt.Sprintf("%.5f", entry.controllerAvg),
+			fmt.Sprintf("%.5f", entry.controllerMax),
+			fmt.Sprintf("%.5f", entry.apiserverAvg),
+			fmt.Sprintf("%.5f", entry.apiserverMax),
+		}
+		if err := w.Write(line); err != nil {
+			klog.Exitf("Error writing data to file; %s", err)
+		}
+	}
+}
+
+func init() {
+	klog.SetOutput(os.Stdout)
+	klog.InitFlags(flag.CommandLine)
+}
+
+func main() {
+	// pull in test variables from flags
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+
+	var plc_filename, output_filename string
+	var n_per_batch, n_total, per_batch_sleep int
+
+	pflag.StringVarP(&plc_filename, "policy", "p",
+		performance_dir+"/resources/templates/deployment-plc.yaml", "path to test policy YAML, scoped to performance directory")
+	pflag.IntVarP(&n_per_batch, "policies-per-batch", "b", 100, "number of policies to create per batch")
+	pflag.IntVarP(&n_total, "total-policies", "t", 1000, "total number of policies created")
+	pflag.IntVarP(&per_batch_sleep, "sleep", "s", 20, "time (min) to sleep after creating a batch of policies")
+	pflag.StringVar(&output_filename, "csv", performance_dir+"results.csv", "path to CSV to export results to")
+
+	pflag.Parse()
+
+	// wait for config map to be applied properly before executing command
+	_, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
```

Comment:
Rather than using `grep`, could you please check in Go (`strings.Contains`) to see if the stdout of the `oc` command contains `prometheus-user-workload-token` .

Also, it'd be nice if you consistently used one Kubernetes CLI rather than go back and forth from `oc` to `kubectl`. If all the commands work with `kubectl`, then I would stick with that.